src/main/java/it/gov/pagopa/ranking/service/RankingRequestsApiServiceImpl.java

```diff
@@ -20,16 +25,18 @@ public class RankingRequestsApiServiceImpl implements RankingRequestsApiService
 
     private final OnboardingRankingRequestsRepository onboardingRankingRequestsRepository;
     private final OnboardingRankingRequest2RankingRequestsApiDTOMapper dtoMapper;
+    private final PageOnboardingRequests2RankingPageDTOMapper pageDtoMapper;
     private final RankingContextHolderService rankingContextHolderService;
 
-    public RankingRequestsApiServiceImpl(OnboardingRankingRequestsRepository onboardingRankingRequestsRepository, OnboardingRankingRequest2RankingRequestsApiDTOMapper rankingRequestsApiDTOMapper, RankingContextHolderService rankingContextHolderService) {
+    public RankingRequestsApiServiceImpl(OnboardingRankingRequestsRepository onboardingRankingRequestsRepository, OnboardingRankingRequest2RankingRequestsApiDTOMapper rankingRequestsApiDTOMapper, PageOnboardingRequests2RankingPageDTOMapper pageDtoMapper, RankingContextHolderService rankingContextHolderService) {
         this.onboardingRankingRequestsRepository = onboardingRankingRequestsRepository;
         this.dtoMapper = rankingRequestsApiDTOMapper;
+        this.pageDtoMapper = pageDtoMapper;
         this.rankingContextHolderService = rankingContextHolderService;
     }
 
     @Override
-    public List<RankingRequestsApiDTO> findByInitiativeId(String organizationId, String initiativeId, int page, int size) {
+    public List<RankingRequestsApiDTO> findByInitiativeId(String organizationId, String initiativeId, int page, int size, String beneficiaryRankingStatus) {
```

Comment:
instead of initialize the out as a ArrayList, let's not initialize it.
Inside the if let's fill it using the stream api (requests.stream().map(....)
in the else, let's return the Collections.emptyList

This way, we could have a performance boost in the population of the lists in case of large size requested